### PR TITLE
Implement based job override on command line

### PR DESCRIPTION
**Pull request by sebastienros**

### CODE REVIEW

**Comment by sebastienros** [comment]
Sorry for the unrelated changes.

Goal is to be able to change what application we want to run. Useful when a scenario uses wrk but we want to use bombardier without creating a duplicate scenario. For instance if you need to use a rate, but wrk doesn't support it. Job properties need to be common though (they are).

File: src/Microsoft.Crank.Controller/Microsoft.Crank.Controller.csproj [hunk 1]
```diff
@@ -10,6 +10,7 @@
     <OutputType>Exe</OutputType>
     <Authors>Microsoft</Authors>
     <PackageId>Microsoft.Crank.Controller</PackageId>
+    <LangVersion>9.0</LangVersion>
```
Comment (davidfowl) [inline src/Microsoft.Crank.Controller/Microsoft.Crank.Controller.csproj:13]: ?

File: src/Microsoft.Crank.Controller/Program.cs [hunk 2]
```diff
@@ -1696,6 +1695,19 @@ int interval
...
+                    foreach (var argument in arguments.ToArray())
+                    {
+                        if (argument.Key == $"{serviceName}.job")
+                        {
+                            jobName = argument.Value;
+
+                            // Remove this argument
+                            arguments = arguments.Where(x => x.Key != argument.Key);
+                        }
+                    }
```
Comment (davidfowl) [inline src/Microsoft.Crank.Controller/Program.cs:1709]: This looks inefficient. 😄

Comment (sebastienros) [reply src/Microsoft.Crank.Controller/Program.cs:1709]: Lame comment.

File: src/Microsoft.Crank.Controller/Configuration.cs [hunk 3]
```diff
@@ -53,7 +53,7 @@ public class Configuration
 
     }
 
-    public class Scenario
+    public class Service
```
Comment (davidfowl) [inline src/Microsoft.Crank.Controller/Configuration.cs:56]: Why this rename?

Comment (sebastienros) [reply src/Microsoft.Crank.Controller/Configuration.cs:56]: when I read my code I was confused, in the yml we have scenarios, but this class represents a named service

**Pull request merged**